### PR TITLE
Enable building winml with --build_nuget

### PR DESCRIPTION
**Pull request by smk2007**
Description: The Microsoft.AI.MachineLearning package cannot be built locally with the --build_nuget parameter. This change enables --build_nuget with the --use_winml flag to produce the Microsoft.AI.MachineLearning.nupkg locally, which contains the winrt API built from source. 

Motivation and Context
Without this change the development and testing of the Microsoft.AI.MachineLearning package is difficult.

### CODE REVIEW

File: csharp/src/Microsoft.AI.MachineLearning.Interop/Microsoft.AI.MachineLearning.props [hunk 1]
```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (ryanlai2) [inline csharp/src/Microsoft.AI.MachineLearning.Interop/Microsoft.AI.MachineLearning.props:1]: Why is this file needed for nuget?

File: tools/ci_build/build.py [hunk 2]
```diff
@@ -1771,9 +1772,13 @@ def build_nuget_package(source_dir, build_dir, configs, use_cuda, use_openvino,
     is_linux_build = derive_linux_build_property()
 
     # derive package name and execution provider based on the build args
+    target_name = "/t:CreatePackage"
     execution_provider = "/p:ExecutionProvider=\"None\""
     package_name = "/p:OrtPackageId=\"Microsoft.ML.OnnxRuntime\""
-    if use_openvino:
+    if use_winml:
+        package_name = "/p:OrtPackageId=\"Microsoft.AI.MachineLearning\""
+        target_name = "/t:CreateWindowsAIPackage"
```
Comment (ryanlai2) [inline tools/ci_build/build.py:1780]: Where is CreateWindowsAIPackage used?

Comment (ryanlai2) [reply tools/ci_build/build.py:1780]: Nevermind, just used as a target name, not references

File: csharp/src/Microsoft.AI.MachineLearning.Interop/Microsoft.AI.MachineLearning.props (hunk 1 above)
Comment (smk2007) [reply csharp/src/Microsoft.AI.MachineLearning.Interop/Microsoft.AI.MachineLearning.props:1]: This is needed so that consumers of the nuget are able to reference the Microsoft.AI.MachineLearning.winmd in their project.

The Samples Gallery uses this to generate a native component that works with the NuGet WinML Types.

**Review by ryanlai2 — APPROVED** [review]
